src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -85,15 +96,20 @@ module Make (Inputs : Inputs_intf) :
     Transition_handler.Validator.run ~frontier
       ~transition_reader:network_transition_reader ~valid_transition_writer
       ~logger ;
+    List.iter collected_transitions
+      ~f:(Strict_pipe.Writer.write primary_transition_writer) ;
+    Strict_pipe.Reader.iter_without_pushback valid_transition_reader
```

Comment:
In that case, I'm worried about this. While we're processing transitions from the pipe backlog, we'll be dropping any "excess" messages here... This shouldn't be a problem in practice I think, we should be able to validate a transition substantially faster than the network is producing them. But if it does happen, I think we will be very confused. It probably _can_ happen in fake_hash, especially given that we still seem to have undiagnosed time leaks.

Last thing I'm concerned about - this never shrinks the pipe capacity back down. How bad is that, @nholland94?